common/rpc/interceptor/namespace_validator.go

```diff
@@ -237,10 +242,35 @@ func (ni *NamespaceValidatorInterceptor) StateValidationIntercept(
 // 4. Namespace from request match namespace from task token, if check is enabled with dynamic config.
 // 5. Namespace is in correct state.
 func (ni *NamespaceValidatorInterceptor) ValidateState(namespaceEntry *namespace.Namespace, fullMethod string) error {
-	if err := ni.checkNamespaceState(namespaceEntry, fullMethod); err != nil {
+	return ni.checkNamespaceState(namespaceEntry, fullMethod)
+}
+
+func (ni *NamespaceValidatorInterceptor) handleRequestWithHandoverRetry(
+	ctx context.Context,
+	req interface{},
+	info *grpc.UnaryServerInfo,
+	handler grpc.UnaryHandler,
+	nsEntry *namespace.Namespace,
+) (interface{}, error) {
+
+	var response interface{}
+	op := func(ctx context.Context) error {
+		err := ni.checkReplicationState(nsEntry, info.FullMethod)
+		if err != nil {
+			var nsErr error
+			nsEntry, nsErr = ni.namespaceRegistry.RefreshNamespaceById(nsEntry.ID())
+			if nsErr != nil {
+				return nsErr
+			}
+			return err
+		}
+
+		response, err = handler(ctx, req)
 		return err
 	}
-	return ni.checkReplicationState(namespaceEntry, fullMethod)
+	// Only retry on ErrNamespaceHandover, other errors will be handled by the retry interceptor
+	err := backoff.ThrottleRetryContext(ctx, op, handoverRetryPolicy, common.IsNamespaceHandoverError)
```

Comment:
I also worry this will cause server api latency alert to fire during the handover, we should probably exclude this backoff from nouserlatency metric.